src/presentations/indexable-post-type-presentation.php

```diff
@@ -74,7 +74,7 @@ public function generate_canonical() {
 			return $this->model->canonical;
 		}
 
-		$permalink = $this->get_permalink();
+		$permalink = ( \get_the_permalink() != $this->get_permalink() ) ? \get_the_permalink() : $this->get_permalink();
```

Comment:
It's same as `$permalink = \get_the_permalink(); ` There is the filter _wpseo_dynamic_permalinks_enabled_ which you can use in this case. If you enable dynamic permalinks then I think that works on this way.